pkg/tfbridge/config_encoding.go

```diff
@@ -218,17 +201,17 @@ func (enc *configEncoding) MarshalProperties(props resource.PropertyMap,
 	return plugin.MarshalProperties(copy, opts)
 }
 
-func (enc *configEncoding) jsonEncodePropertyValue(k resource.PropertyKey,
+func (enc *ConfigEncoding) jsonEncodePropertyValue(k resource.PropertyKey,
 	v resource.PropertyValue) (resource.PropertyValue, error) {
 	if v.ContainsUnknowns() {
 		return resource.NewStringProperty(plugin.UnknownStringValue), nil
 	}
-	if v.IsSecret() {
-		encoded, err := enc.jsonEncodePropertyValue(k, v.SecretValue().Element)
+	if v.ContainsSecrets() {
```

Comment:
This will encode any nested secrets as top-level secret. Because it can't represent nested secrets accurately in the JSON-in-proto encoding.